tests/testthat/test-write-permissions.R

```diff
@@ -0,0 +1,14 @@
+
+context("error without write permissions")
+
+test_that("test failed write errors for saveWorkbook", {
+  tempFile <- fs::file_temp(ext = "xlsx")
+  fs::file_create(tempFile, mode = "a=rx")
```

Comment:
Could this be accomplished without important `fs`?

```r
tempFile <- temp_xlsx()
file.create(tempFile)
Sys.umask(file.mode(tempFile))
```

Reply:
Unfortunately, `file.create` doesn't allow for setting permissions, so I chose to use `fs::file_create` instead of using `file.create` and then having platform-specific commands to set the read-only status.

Reply:
I am not sure if it is necessary to have a new dependency for a function used in the test.

Reply:
Should I remove the dependency and just have the GitHub workflows install the fs package explicitly for those which run the tests?

Reply:
I think I found another way. I'll update the branch later this evening.

Reply:
Fixed in ec229fea1de5033c9cf25baebe0a019a07717998